src/test/java/CalendarServletTest.java

```diff
@@ -76,54 +95,75 @@ public void setUp() throws Exception {
     servlet = new CalendarServlet(authenticationVerifier, calendarClientFactory);
 
     Mockito.when(calendarClientFactory.getCalendarClient(Mockito.any())).thenReturn(calendarClient);
+    // Authentication will always pass
+    Mockito.when(authenticationVerifier.verifyUserToken(Mockito.anyString())).thenReturn(true);
+
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+
+    request = Mockito.mock(HttpServletRequest.class);
+    response =
+        Mockito.mock(
+            HttpServletResponse.class,
+            AdditionalAnswers.delegatesTo(new HttpServletResponseFake(stringWriter)));
+    Mockito.when(request.getCookies()).thenReturn(validCookies);
   }
 
   @Test
   public void noCalendarEvent() throws Exception {
     // Test case where there are no events in the user's calendar
+    // The result should be that all hours are free
     Mockito.when(calendarClient.getCalendarList()).thenReturn(ONE_CALENDAR);
-    Mockito.when(calendarClient.getCalendarEvents(PRIMARY)).thenReturn(NO_EVENT);
-    List<Event> actual = getServletResponse();
-    Assert.assertEquals(NO_EVENT, actual);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(NO_EVENT);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    CalendarSummaryResponse actual = getServletResponse();
+    List<Integer> workHoursPerDay = new ArrayList<Integer>(Collections.nCopies(5, 8));
+    List<Integer> personalHoursPerDay = Arrays.asList(6, 8, 8, 8, 8);
+    Assert.assertEquals(2, actual.getStartDay());
+    Assert.assertEquals(5, actual.getWorkTimeFree().size());
+    Assert.assertEquals(5, actual.getPersonalTimeFree().size());
+    for (int index = 0; index < workHoursPerDay.size(); index++) {
+      Assert.assertEquals(
+          (int) workHoursPerDay.get(index), actual.getWorkTimeFree().get(index).getHours());
+      Assert.assertEquals(0, actual.getWorkTimeFree().get(index).getMinutes());
+      Assert.assertEquals(
+          (int) personalHoursPerDay.get(index), actual.getPersonalTimeFree().get(index).getHours());
+      Assert.assertEquals(0, actual.getPersonalTimeFree().get(index).getMinutes());
+    }
   }
 
   @Test
   public void twoCalendars() throws Exception {
     // Test case where there are two calendars with a defined event in each
-    // Events must be returned in order of retrieval
+    // Event with 1 working hour and event with 2 personal hours.
```

Comment:
This comment is a little misleading because the 2nd event only overlaps with personal time for 1 hour. I think the test is actually correct, and the comment just needs to be updated.

Reply:
Thank you, `Resolved`